### ISSUE TITLE
Consider splitting/rebranding library

### ISSUE THREAD

**Issue by axelfontaine** [kotlindev/underscore-kotlin#22]
With the Kotlin standard library, there is a significant overlap in functionality between the built-in collections and what this library offers. 

From https://github.com/kotlindev/underscore-kotlin?tab=readme-ov-file#usage

This is fully covery by the Kotlin standard library:

```kotlin
U.of(/* array | list | set | map | anything based on Iterable interface */)
    .filter(..)
    .map(..)
    ...
    .sortWith()
    .forEach(..);
U.of(value1, value2, value3)...
U.range(0, 10)...

U.of(1, 2, 3) // or java.util.Arrays.asList(1, 2, 3) or new Integer[] {1, 2, 3}
    .filter(v -> v > 1)
    // 2, 3
    .map(v -> v + 1)
    // 3, 4
    .sortWith((a, b) -> b.compareTo(a))
    .forEach(System.out::println);
    // 4, 3
    
U.of(1, 2, 3) // or java.util.Arrays.asList(1, 2, 3) or new Integer[] {1, 2, 3}
    .mapMulti((num, consumer) -> {
        for (int i = 0; i < num; i++) {
            consumer.accept("a" + num);
        }
    })
    .forEach(System.out::println);
    // "a1", "a2", "a2", "a3", "a3", "a3"
```

And this can be expressed more clearly and concisely with nested `mapOf()`and `listOf()` method calls:

```kotlin
U.Builder builder = U.objectBuilder()
    .add("firstName", "John")
    .add("lastName", "Smith")
    .add("age", 25)
    .add("address", U.arrayBuilder()
        .add(U.objectBuilder()
            .add("streetAddress", "21 2nd Street")
            .add("city", "New York")
            .addNull("cityId")
            .add("state", "NY")
            .add("postalCode", "10021")))
    .add("phoneNumber", U.arrayBuilder()
        .add(U.objectBuilder()
            .add("type", "home")
            .add("number", "212 555-1234"))
        .add(U.objectBuilder()
            .add("type", "fax")
            .add("number", "646 555-4567")));
```

Same here:

```kotlin
XmlBuilder builder = XmlBuilder.create("Projects")
    .e("underscore-kotlin").a("language", "Java").a("scm", "SVN")
        .e("Location").a("type", "URL")
            .t("https://github.com/kotlindev/underscore-kotlin/")
        .up()
    .up()
    .e("JetS3t").a("language", "Java").a("scm", "CVS")
        .e("Location").a("type", "URL")
            .t("https://jets3t.s3.amazonaws.com/index.html");
```

---

**However what Kotlin is completely lacking is easy and lightweight JSON and XML processing.**

It might be worth considering extracting the JSON <-> Kotlin Collections <-> XML part into its own library, implemented in pure Kotlin. This would be immensely useful for both Kotlin/JVM and multi-platform projects. 

The JSON <-> Kotlin Collections part is already almost pure Kotlin. Only the Kotlin Collections <-> XML would have to be rewritten to drop the JVM dependency.

You could then establish this new library as the go-to lightweight solution for JSON and XML in the entire Kotlin ecosystem. 

In short: **less code, more focus, higher reach and clearer positioning**.

Just an idea.

**Comment by javadev** [comment]
Hi @axelfontaine,

Thanks for bringing up the issue.

The library was developed as a counterpart to underscore-java. The XML component currently utilizes JVM to read and generate.
